persistence-api/src/main/java/io/stargate/db/RowDecorator.java

```diff
@@ -29,4 +30,6 @@
    * same order that queries iterate / paginate over the Cassandra data ring.
    */
   <T extends Comparable<T>> ComparableKey<T> decoratePartitionKey(Row row);
+
+  Stream<Byte> getComparableBytes(Object... rawKeyValues);
```

Comment:
I decided to use Stream<Byte> here because `ByteSource` (the class that comparable bytes returns) is best represented as an iterable of bytes, rather than a list with known length - so the consumer of this method would be responsible for either consuming the whole stream and creating a list, or passing the stream along (maybe gRPC supports this?)

Reply:
But isn't `Stream<Byte>` very inefficient? `Byte` is wrapper type and that seems ... like a very inefficient abstraction, and at least wrt performance worse than `ByteSource`.

Reply:
ah I should use ByteBuffer 🤦 idk what I was thinking

Reply:
Yeah that's a solid choice. Np! :)

Reply:
Yep good points @tatu-at-datastax